source/core/Native.cs

```diff
@@ -0,0 +1,711 @@
+            }
+
+            // String types
+            public static implicit operator InputArgument(string value)
+            {
+                return new InputArgument(value);
+            }
+            public static unsafe implicit operator InputArgument(char* value)
+            {
+                return new InputArgument(new string(value));
+            }
+
+            // Pointer types
+            public static implicit operator InputArgument(IntPtr value)
+            {
+                return new InputArgument(value);
+            }
+            public static unsafe implicit operator InputArgument(void* value)
+            {
+                return new InputArgument(new IntPtr(value));
+            }
+
+            public static implicit operator InputArgument(OutputArgument value)
+            {
+                return new InputArgument(value._storage);
+            }
+        }
+
+        public class OutputArgument
+        {
+            internal IntPtr _storage = Marshal.AllocCoTaskMem(24);
+            private bool disposed;
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="OutputArgument"/> class for natives that output data into pointers.
+            /// </summary>
+            public OutputArgument()
+            {
+            }
+
+            /// <summary>
+            /// Initializes a new instance of the <see cref="OutputArgument"/> class with an initial value for natives that require the pointer to data instead of the actual data.
+            /// </summary>
+            /// <param name="initvalue">The value to set the data of this <see cref="OutputArgument"/> to.</param>
+            public OutputArgument(object value) : this()
+            {
+                unsafe
+                {
+                    *(ulong*)(_storage) = Function.ObjectToNative(value);
+                }
+            }
+
+            public void Dispose()
+            {
+                Dispose(true);
+                GC.SuppressFinalize(this);
+            }
+            protected virtual void Dispose(bool disposing)
+            {
+                if (!disposed)
+                {
+                    if (disposing)
+                    {
+                    }
+                    Marshal.FreeCoTaskMem(_storage);
+                    disposed = true;
+                }
+            }
+            ~OutputArgument()
+            {
+                Dispose(false);
+            }
+
+            /// <summary>
+            /// Gets the value of data stored in this <see cref="OutputArgument"/>.
+            /// </summary>
+            public T GetResult<T>()
+            {
+                unsafe
+                {
+                    return (T)(Function.ObjectFromNative(typeof(T), (ulong*)(_storage.ToPointer())));
+                }
+            }
+        }
+
+        public sealed class Function
+        {
+            internal unsafe struct NativeTask : IScriptTask
```

Comment:
This should be a `class`. Both C++/Cli `ref struct` and `ref class` equal `class` in C#, `value struct` and `value class` equal struct in C#.

Reply:
I didn't define this as a `class` for some reason. I know what you're saying, yes.